app/src/main/java/com/android/bookswap/ui/map/Map.kt

```diff
@@ -104,17 +107,36 @@ fun MapScreen(
 
   var mutableStateSelectedUser by remember { mutableStateOf(selectedUser) }
   var markerScreenPosition by remember { mutableStateOf<Offset?>(null) }
-  val listAllBooks = listUser.flatMap { it.listBook }
+  val userBooksList = remember { mutableStateListOf<UserBooksWithLocation>() }
+
+  // Fetch books
+  LaunchedEffect(Unit) {
+    booksRepository.getBook(
+        OnSucess = { books ->
+          listUser.forEach { user ->
+            userBooksList.add(
+                UserBooksWithLocation(
+                    user.longitude,
+                    user.latitude,
+                    books.filter { book -> book.uuid in user.bookList }))
+          }
+        },
+        onFailure = {})
```

Comment:
could provide more information if there is an error with example an Log

Reply:
I will do it in another PR as I am currently working on a branch that remove the logic for getting the list of books from the UI and put it in a model file.